shopBuddy/shopBuddy/ShoppingListManagerViewController.m

```diff
@@ -0,0 +1,78 @@
+//
+//  ShoppingListManagerViewController.m
+//  shopBuddy
+//
+//  Created by Rachna Gupta on 7/18/22.
+//
+
+#import "ShoppingListManagerViewController.h"
+#import "ShoppingListCell.h"
+#import "ShoppingList.h"
+#import "SpecificListViewController.h"
+#import "ShoppingList+Persistent.h"
+#import "Parse/Parse.h"
+@interface ShoppingListManagerViewController () <UITableViewDataSource, UITableViewDelegate>
+{
+    __weak IBOutlet UITableView *tableView;
+    NSArray<ShoppingList*> *lists;
+    
+}
+
+@end
+
+@implementation ShoppingListManagerViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    tableView.dataSource = self;
+    tableView.delegate = self;
+    tableView.rowHeight = UITableViewAutomaticDimension;
+    __weak __typeof__(self) weakSelf = self;
+    [weakSelf reloadLists];
+}
+
+- (void)reloadLists {
+    [ShoppingList fetchListsByUser:[PFUser currentUser] withCompletion:^(NSArray<ShoppingList *> *lists, NSError *error) {
+        self->lists = lists;
+        [self->tableView reloadData];
+    }];
+}
+- (IBAction)didTapAddList:(id)sender {
+    //TODO: add List names (make List of stores)
+    __weak __typeof__(self) weakSelf = self;
+    [ShoppingList createEmptyList: @"New_List" withCompletion:^(ShoppingList *list,NSError *error) {
+        if(!error) {
+            [weakSelf reloadLists];
```

Comment:
Can you just append the new list to the local copy `lists`, instead of reloading all.